src/scripts/components/Notice.js

```diff
@@ -9,38 +9,38 @@ import { NoticeIcon } from './NoticeImage';
 import { NoticeActions } from './NoticeAction';
 
 // Import utilities
+// @ts-ignore
 import classnames from 'classnames';
 import { purify } from '../utils/sanitization';
-import moment from 'moment';
-import { __ } from '@wordpress/i18n';
 import { defaultContext, NOTIFY_NAMESPACE } from '../store/constants';
 import { dispatch } from '@wordpress/data';
+import { NoticeMeta } from './NoticeMeta';
+import { delay } from '../utils';
 
-export const NoticeMeta = ( { date, source } ) => (
-	<p className="wp-notification-meta">
-		<span className="name">{ source }</span> { '\u2022 ' }
-		<span className="date">{ moment( date ).fromNow() }</span>
-	</p>
-);
+/**
+ * @typedef {import('../store').Notice} Notice
+ */
 
 /**
  * It renders a single notice
  *
- * @param {Object} props
+ * @param {Notice} props
  * @return {JSX.Element} Notice - the single notice
  */
 export const Notice = ( props ) => {
 	const {
-		id,
-		title,
-		status,
+		action,
 		context = defaultContext,
-		source = 'WordPress',
-		date = __( 'Just now' ),
+		date = Date.now() * 0.001,
```

Comment:
```suggestion
		date = Math.floor(Date.now() / 1000),
```

The date should be an integer.